### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by magalimorin18**

### CODE REVIEW

File: test/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      expect(signedObject).toHaveProperty("r");
+      expect(signedObject).toHaveProperty("s");
+      expect(signedObject).toHaveProperty("signature");
+    });
+  });
+});
+describe("Recover the address function of a transaction", () => {
+  it("should recover the signing address of a transaction", () => {
+    const lsp6Signer = new LSP6Signer();
+    const signingAddress = "0x2b389f8EB52D16A105e02165a2AC1450461A237b";
```
Comment (CallumGrindle) [inline test/index.test.ts]: this can go with the global `signingKey` param as they are related and will not change

File: test/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    dataTest.forEach((data) => {
+      const messageInfo = lsp6Signer.sign(data, signingKey);
```
Comment (CallumGrindle) [inline test/index.test.ts]: `messageData` maybe?

File: test/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        isMessagePrefixed
+      );
+      expect(recoveredAddress).toBe(signingAddress);
+      expect(recoveredAddress.substring(0, 2)).toBe("0x");
+      expect(recoveredAddress.length).toBe(42);
+    });
+  });
+  it("should not be equal because the message is not prefixed", () => {
+    const lsp6Signer = new LSP6Signer();
+    const signingAddress = "0x2b389f8EB52D16A105e02165a2AC1450461A237b";
```
Comment (CallumGrindle) [inline test/index.test.ts]: This is the same as above, by moving it to the global context of this file we can reuse it without having to create it twice in different contexts

File: test/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        message,
+        signature,
+        isMessagePrefixed
+      );
+      expect(recoveredAddress).toBe(signingAddress);
+      expect(recoveredAddress.substring(0, 2)).toBe("0x");
+      expect(recoveredAddress.length).toBe(42);
+    });
+  });
+  it("should not be equal because the message is not prefixed", () => {
```
Comment (CallumGrindle) [inline test/index.test.ts]: Can we also check that the result is equal when passing a prefixed vs not prefixed version of the message with the correct `isMessagePrefixed` param? e.g. passing 'hello' with isMessagePrefixed=false vs passing prexif + 'hello' with isMessagePrefixed=true should both give the correct addresss

File: test/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      const isMessagePrefixed = true;
+      const recoveredAddress = lsp6Signer.recover(
+        message,
+        signature,
+        isMessagePrefixed
+      );
+      expect(recoveredAddress).not.toBe(signingAddress);
+    });
+  });
+});
```
Comment (CallumGrindle) [inline test/index.test.ts:152]: I think we can also improve spacing in this file so it is more readable. Atm it is a block of text which can make it harder to read

**Review by CallumGrindle — CHANGES_REQUESTED** [review]
Cool. Please see comments

**Review by Hugoo — CHANGES_REQUESTED** [review]
This PR has commits not related to this PR, but commits from this PR:

image 
https://github.com/lukso-network/tools-lsp6-signer/pull/3/commits

I think you should rebase or clean this branch so it only has the commits relevant to this PR.

**Comment by Hugoo** [comment]
I suggest to use `kebab-case` for branches names

File: test/index.test.ts [hunk 6]
```diff
@@ -1,51 +1,103 @@
...
+      );
+      expect(recoveredAddress).toBe(signingAddress);
+      expect(recoveredAddress.substring(0, 2)).toBe("0x");
+      expect(recoveredAddress.length).toBe(42);
+    });
   });
-  it("should be hexadecimal", () => {
-    expect(recoveredAddress.substring(0, 2)).toBe("0x");
+
+  it("should recover the signing address of a transaction when the message is prefixed", () => {
```
Comment (CallumGrindle) [inline test/index.test.ts]: This is not really when the message is prefixed, as the has function is doing the prefixing for us here. It is more like 'should recover the signer address when recovering signed hash'

File: test/index.test.ts [hunk 7]
```diff
@@ -1,51 +1,103 @@
 import { LSP6Signer } from "../src/index";
-
 const signingKey =
   "ffeb17b9a6059fec3bbab63d76b060b7380cac7a62ce6621a134531a46458968";
 
+const signingAddress = "0x2b389f8EB52D16A105e02165a2AC1450461A237b";
+
+const dataTest = [
```
Comment (CallumGrindle) [inline test/index.test.ts]: maybe `testCases` is a better variable name?

**Review by CallumGrindle — APPROVED** [review]

File: test/index.test.ts [hunk 8]
```diff
@@ -3,49 +3,102 @@ import { LSP6Signer } from '../src/index';
...
+  'amuse',
+  'breakfast',
+  'series',
+  'season',
+  'recovery',
+  'threshold',
+  'step',
+  'composer',
+  'mill',
+  'verdict',
```
Comment (Hugoo) [inline test/index.test.ts]: I think you can be more exotic in your test cases - add sentences, add hex strings, numbers, etc.

**Review by Hugoo — CHANGES_REQUESTED** [review]
I think you should add a test that checks if the prefix is: `\x19LSP6 Execute Relay Call:\n`.

It looks like if I go to the index, change the prefix to whatever, the tests will still pass. Or?

File: test/index.test.ts [hunk 9]
```diff
@@ -101,4 +98,41 @@ describe('Recover the address function of a transaction', () => {
       expect(recoveredAddress).not.toBe(signingAddress);
     });
   });
+
+  it('should recover the adress even if the message is an object', () => {
```
Comment (CallumGrindle) [inline test/index.test.ts]: small typo. adress -> address

File: test/index.test.ts [hunk 10]
```diff
@@ -101,4 +98,41 @@ describe('Recover the address function of a transaction', () => {
...
+      const recoveredAddress = lsp6Signer.recover(
+        messageData,
+        signature,
+        isMessagePrefixed,
+      );
+      expect(recoveredAddress).toBe(signingAddress);
+    });
+  });
+
+  it('should be prefixed with "\x19LSP6 Execute Relay Call:\n"', () => {
```
Comment (CallumGrindle) [inline test/index.test.ts]: this is more like testing 'should accept both hashed and unhashed message'

File: test/index.test.ts [hunk 11]
```diff
@@ -101,4 +98,41 @@ describe('Recover the address function of a transaction', () => {
...
+    const lsp6Signer = new LSP6Signer();
+
+    const rawData = 'hello';
```
Comment (CallumGrindle) [inline test/index.test.ts]: You can use the same testCases.forEach pattern here so we cover more cases

Comment (Hugoo) [reply test/index.test.ts]: Like here:
https://github.com/ERC725Alliance/erc725.js/blob/develop/src/index.test.ts#L385
https://github.com/ERC725Alliance/erc725.js/blob/develop/src/index.test.ts#L1039

File: test/index.test.ts (hunk 10 above)
Comment (Hugoo) [reply test/index.test.ts]: If i read this in the title, I would expect to see a test that has "expect LSP6 Execute Relay Call..." somewhere. I don't see it.

**Review by CallumGrindle — APPROVED** [review]

File: test/index.test.ts [hunk 12]
```diff
@@ -86,7 +87,7 @@ describe('Recover the address function of a transaction', () => {
     });
   });
 
-  it('should recover the wrong signing adress as isMessagePrefixed is set to true and the message is not prefixed ', () => {
+  it('should recover the wrong signing adress when the message is not prefixed but isMessagePrefixed is set to true', () => {
```
Comment (Hugoo) [inline test/index.test.ts]: maybe:
it should no recover the expected address xxx

**Review by Hugoo — APPROVED** [review]

**Pull request merged**